website/docs/transforms/mock.md

```diff
@@ -26,6 +26,144 @@ transforms:
 
 The example above will replace the resolver of `User.firstName` with a mock that uses `faker.js` to generate a random name.
 
+## Custom mock functions for types
+
+You can mock types with custom mock functions like below;
+
+```yml
+transforms:
+  - mock:
+      mocks:
+        - apply: DateTime
+          custom: graphql-scalars#DateTimeMock 
+          # This will import `DateTimeMock` from `graphql-scalars` for example. Local paths are also supported
+```
+
+## Custom mock functions for fields
+
+You can mock a specific field of a type;
+
+```yml
+transforms:
+  - mock:
+      mocks: 
+        - apply: User.fullName
+          custom: ./user-mocks#fullName
+```
+
+in `user-mocks.js` file;
+
+```js
+module.exports = {
+  fullName: () => 'My Static Name'
+};
+```
+
+## Mocking the lists
+
+By default, Mesh generates two mocked items if the return type is a list. But this can be configured;
+
+Let's say, you have a schema like below;
+
+```graphql
+  type Query {
+    users: [User]
+  }
+  type User {
+    id: ID
+    fullName: String
+  }
+```
+
+Then a configuration like below;
+
+```yml
+transforms:
+  - mock:
+      mocks: 
+        - apply: User.fullName
+          custom: ./user-mocks#fullName
```

Comment:
Would it be possible to provide specific values per item in an array?

Reply:
You can return a function that randomly gives the values;
```js
const items = ['John Doe', 'Jane Doe'];
module.exports.fullName = () =>  items[Math.floor(Math.random() * items.length)].
```

Reply:
Could mocks be populated with data in a predictable manner?

For example, for more complex data models, e.g. `user`:
```ts
{
  id: ID!,
  name: String!,
  email: AppropriateEmailType,
  age: Int,
  school: String,
  etc.
}
```
It would be nice if the list of users can be provided in such a way that when queried it always returns a predictable result.

Reply:
Yes you can provide the mocks in the same way;
`apply: User`
 Then select a random mock result randomly from the array

Reply:
Okay. 
Is there an example of how that might look like?
I would expect something like this:

```ts
// mocks.ts
const items = [
  {
    id: 'id-val-1',
    name: 'John Doe',
    ...
  },
  {
    id: 'id-val-2',
    name: 'John Snow',
    ...
  }
]

export default users
```
```yaml
  - mock:
      mocks: 
        - apply: User
          custom: ./user-mocks
```

Reply:
By combining your suggested approach with some generator magic I've made it work :)

```ts
function* generateNames() {
  yield 'John Doe';
  yield 'John Snow';
  yield 'Me';
  yield 'You';
  yield 'Too';
}

const name = generateNames();
module.exports = {
  name: () => name.next().value,
};
```